components/scream/src/control/surface_coupling.cpp

```diff
@@ -69,7 +69,8 @@ set_num_fields (const int num_cpl_imports, const int num_scream_imports,
 void SurfaceCoupling::
 register_import(const std::string& fname,
                 const int cpl_idx,
-                const int vecComp)
+                const int vecComp,
+                const bool flip_sign)
```

Comment:
Note: whether Up=1 and Down=-1 or the other way around is irrelevant. The point is that Up*Down=-1, so that whenever the two are different, you get the sign flip, otherwise no sign flip.

The important thing is to flag both cpl/scr fluxes as "Up" if they point in the same direction.

Reply:
@AaronDonahue can correct me if I misheard, but I believe for exporting (scr -> cpl), the surface models wants scream to hand back fluxes in scream's convention, and then it does the `*-1` internally. So then SurfaceCoupling only needs to do this stuff for import, which will be a bit annoying/confusing.

Anyways, I agree that the way I've done it is not ideal. I'll work on a better design.

Reply:
@bartgol Even with this array approach, I think I will still have to pass in whether or not we flip the sign into `register_import()` since `SurfaceCoupling` does not know the ordering of the imports beforehand. Either that, or inside `register_import()` I can query the field name and decide then if it has an up or down direction manually.

Reply:
I mean, the way you did it works, and it's likely going to be fine. But if someone changes the sign of scream fluxes later on, we'll get errors, with little knowledge to what happened, and why. Having a var for explicitly stating the sign of fluxes can help driving the debugging. It's a documentation of sort, built in the code. We could do something simpler:

```
// Store 1 for fluxes with same convention between cpl and scream, and -1 otherwise
constexpr std::array<int,NUM_IMPORTS> cpl_scr_flux_sign_flip = { 1, 1, 1, -1, 1, 1, -1, ... };
```
And later on just do
```
  info.data[offset] = cpl_imports_view_d(...) * cpl_scr_flux_sign_flip[ifield];
```
That would also allow you to not modify the register_import signature, since the flip sign would be hard coded in C++.

Reply:
Could it be a hybrid? I.e., `register_import()` take in the `cpl_flux_dir` as an input and `SurfaceCoupling` hard code it's own `scr_flux_dir`, then those are combined in `do_import()`.

A thing that makes this a bit awkward is that some imports are fluxes, and then others are these rrtmgp albedo fields, like `sfc_alb_visr`, which are not directional. So we'll have to store a "flux direction" for a non flux field.

Reply:
Ah, right, SurfaceCoupling does not know the order of the imports. And you're right, some fields are not fluxes. Annoying.

If I understand correctly, we do need this fix soon to get the model to work correctly. Then I guess we can polish the design in a second PR (possibly, not far in the future). Feel free to get the code working in whichever way you prefer for now.

Reply:
I went with a hardcoded/simplified version similar to 3 comments above, with a TODO that says this should be improved. It removes the need for changing the `register_import()` signature or changing any mct code.